File: Service Lane/ServiceLane.py
'''
Created on 24/07/2014

@author: miguel

    Sample Input #00
    
    8 5
    2 3 1 2 3 2 3 3
    0 3
    4 6
    6 7
    3 5
    0 7
    Sample Output #00
    
    1
    2
    3
    2
    1

'''
import sys


if __name__ == '__main__':
    
    lengthFreeway, numberOfCases = map(int, sys.stdin.readline().split())
    nextLane = str(sys.stdin.readline())
    lane = list("".join(nextLane).split())
    for i in range(numberOfCases):
        enterLane, exitLane = map(int, sys.stdin.readline().split())
        minLane = 3
        for j in range(enterLane, exitLane+1):
            minLane = min(int(lane[j]), minLane)
        print(minLane)
                

File: Service Lane/ServiceLane.java
import java.util.Scanner;


/*  8 5
 *	2 3 1 2 3 2 3 3
 *	0 3
 *	4 6
 *	6 7
 *	3 5
 *	0 7
 */

public class ServiceLane {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		
		int lengthFreeway = in.nextInt();
		int testCases = in.nextInt();
		
		int[] lane = new int[lengthFreeway];
		for(int i = 0; i < lengthFreeway; i++){
			lane[i] = in.nextInt();
		}
		for(int i = 0; i < testCases; i++){
			int enterLane = in.nextInt();
			int exitLane = in.nextInt();
			int minLength = 3;
			for(int j = enterLane; j <= exitLane; j++){
				if(lane[j] == 1){
					if(minLength > 1)
						minLength = 1;
					}
				else if(lane[j] == 2){
						if(minLength > 2)
							minLength = 2;
						}
			}
			System.out.println(minLength);
		}
	}

}
